Dockerfile

```diff
@@ -1,36 +1,47 @@
-FROM node:10.16.2-stretch
+FROM node:10-buster
 
-RUN \
-  apt-get update && \
-  apt-get -y upgrade && \
-  apt-get install -y build-essential && \
-  apt-get install -y software-properties-common && \
-  apt-get install -y curl git man unzip vim python \
-    ca-certificates file autoconf automake autotools-dev libtool \
-    xutils-dev
-
-# Rust stuff
-ENV SSL_VERSION=1.0.2s
-
-RUN curl https://www.openssl.org/source/openssl-$SSL_VERSION.tar.gz -O && \
-    tar -xzf openssl-$SSL_VERSION.tar.gz && \
-    cd openssl-$SSL_VERSION && ./config && make depend && make install && \
-    cd .. && rm -rf openssl-$SSL_VERSION*
-
-ENV OPENSSL_LIB_DIR=/usr/local/ssl/lib \
-    OPENSSL_INCLUDE_DIR=/usr/local/ssl/include \
-    OPENSSL_STATIC=1
-
-# install toolchain
-RUN curl https://sh.rustup.rs -sSf | \
-    sh -s -- --default-toolchain nightly -y
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain nightly -y
 
 ENV PATH=/root/.cargo/bin:$PATH
-RUN rustup override set nightly-2019-07-04
-
-WORKDIR /root
-COPY /etc/start-build.sh start-build.sh
-RUN npm install -g yarn
 
-ENTRYPOINT ["bash"]
-CMD ["/root/start-build.sh"]
\ No newline at end of file
+COPY rust-toolchain /root/salad/rust-toolchain
+
+RUN export RUST_NIGHTLY=$(cat /root/salad/rust-toolchain) && \
+  rustup toolchain install $RUST_NIGHTLY && \
+  rustup target add wasm32-unknown-unknown --toolchain $RUST_NIGHTLY
```

Comment:
This dockerfile is pretty heavyweight. I think you can optimize it by adding multiple stages to the build. You can see this in the dockerfiles I introduce in https://github.com/enigmampc/docker-environment/pull/19 , although if this is only a temporary solution maybe it doesn't really matter. (I'm assuming the CI caches some of the intermediate docker images if it doesn't, your solution is probably more efficient)

Reply:
Correct. I am aware of it, and the multiple stage builds. I don't expect anyone else to use this image, because developers can use this repo outside any container just by launching the network and connecting to it. I don't expect to distribute this image, only the CI to build it locally and use it for testing, so it's optimized for build speed towards that end.

Reply:
Fair enough, sounds reasonable :)